global/scripts/debug.shlib

```diff
@@ -0,0 +1,275 @@
+#!/bin/echo This script should not run directly:
+
+# Description:
+#
+#   Function for printing debugging information on caught errors.
+#   It makes use of error trapping. 
+#
+#   To disable error trapping run `debug_disable_trap`
+#   To disable exit uppon trapped error run `debug_disable_exit`
```

Comment:
Typo: Uppon

Reply:
Also, can we perhaps expand the comments further, for users looking for a simple understanding of how this all work without delving into the code?  e.g.,
```
#   Function for printing debugging information on caught errors.
#   It makes use of error trapping. 
#
#   Default behavior is that error trapping is enabled, 
#   and the script exits upon trapped error with the same status as the failed command.
#   
#   To disable error trapping altogether run `debug_disable_trap`
#      (Note that if `set -e` is set in the calling script, the script would still exit upon error 
#       through that mechanism).
#   To disable exit upon trapped error run `debug_disable_exit`
#       In this case (trap doesn't exit), execution proceeds to the next line of the script, 
#       with `$?` still set to whatever value the failed command errored with.
```

Reply:
fixed and expanded.